common/story_cycles/wc_story_cycle_thunder_king.txt

```diff
@@ -0,0 +1,159 @@
+﻿story_thunder_king = { # Thunder King Crisis by Juke
+	# The Thunder King Lei Shen is resurrected and begins his reconquest of Pandaria
+	# Can be done (By player and, less likely, by AI) via decision, or it can trigger randomly (see gamerule)
+	# https://wowpedia.fandom.com/wiki/Lei_Shen
+	# Juke note: I originally handled this through on_actions and hidden events, hence a lot of effects are contained in events and scripted effects
+	# This makes it a bit harder to get an overview of all the events, but makes debugging a bit easier
+	# If this is broken in the future or something message me on discord: juketm
+	
+	### See related files:
+	# common/decisions/wc_major_pandaria_decisions.txt
+	# common/on_action/game_start.txt
+	# common/on_action/yearly_on_actions.txt
+	# common/game_rules/wc_game_rules.txt
+	# common/scripted_character_templates/wc_mogu_templates.txt
+	# common/scripted_effects/wc_pandaria_effects.txt
+	# events/decisions_events/wc_major_pandaria_events.txt
+	
+	### Can be initiated through:
+	# - yearly_on_action
+	# - revive_lei_shen_decision
+	# - zandalari_lei_shen_decision
+	
+	on_setup = {
+		if = { # Make sure lightning don't strike twice
+			limit = {
+				NOT = {
+					is_target_in_global_variable_list = {
+						name = unavailable_unique_decisions
+						target = flag:revive_lei_shen_decision
+					}
+				}
+			}
+			add_to_global_variable_list = {
+				name = unavailable_unique_decisions
+				target = flag:revive_lei_shen_decision
+			}
+		}
+		story_owner = {
+			save_scope_as = lei_shen
+			var:reviver ?= {
+				save_scope_as = reviver
+			}
+		}
+		
+		# Handles Court setup, troops, etc
+		story_owner = {
+			trigger_event = pandaria_unification.0101
+		}
+		
+		## Handles Mogu-Zandalari Pact
+		#if = {
+		#	limit = {
+		#		scope:reviver.culture = { has_cultural_pillar = heritage_zulite }
+		#	}
+		#	story_owner = {
+		#		trigger_event = {
+		#			id = pandaria_unification.0302
+		#			days = 10
+		#		}
+		#	}
+		#	scope:reviver = {
+		#		trigger_event = {
+		#			id = pandaria_unification.0302
+		#			days = 10
+		#		}
+		#	}
+		#}
+		
+	}
+	
+	# Disband his realm, sends news event
+	on_end = {
```

Comment:
is there any other way besides Lei Shen's death for this story cycle to end?

if not, and he's still getting these events, then why is the crisis considered to be over if he's still alive but his title has been usurped and he's not an emperor anymore?